src/main/java/uk/gov/hmcts/darts/common/repository/TransformedMediaRepository.java

```diff
@@ -58,4 +59,30 @@ AND upper(tod.status.description) <> 'MARKED FOR DELETION'
         """)
     List<TransformedMediaEntity> findAllDeletableTransformedMedia(OffsetDateTime createdAtOrLastAccessedDateTime);
 
-}
+    @Query("""
+        SELECT tm
+            FROM TransformedMediaEntity tm
+            JOIN tm.mediaRequest media
+            JOIN media.hearing hearing
+            JOIN hearing.courtCase courtCase
+            JOIN hearing.courtroom courtroom
+            JOIN courtroom.courthouse courthouse
+        WHERE
+           (:mediaId IS NULL OR (:mediaId IS NOT NULL AND media.id=:mediaId)) AND
+           (:caseNumber IS NULL  OR (:caseNumber IS NOT NULL AND courtCase.caseNumber=:caseNumber)) AND
+           (:courtHouseDisplayName IS NULL OR (:courtHouseDisplayName IS NOT NULL AND courthouse.displayName ILIKE CONCAT('%', :courtHouseDisplayName, '%'))) AND
```

Comment:
this may not be too important in this scenario because the tables are small, but i believe ilike does not make use of indexes.
we have put UPPER indexes on some columns in the db where we know we will frequently do case-insensitive searches on large tables, and so when you do something lke:-
`UPPER(case_number) = upper(:casenumber)`

it will make use of the UPPER index.
I think using an ILIKE does not use any index, unless the database has a certain feature installed:-
https://stackoverflow.com/questions/20336665/lower-like-vs-ilike

When you use JPA to generate an IgnoreCase query, it generates it with UPPER, so to remain consistent, we have used UPPER indexes everywhere

Reply:
i'm asking Barry about the GIN indexes to see what he thinks

Reply:
let me know if I need to change this to use UPPER